### PR TITLE
Update Developer guide and DevOps guide

### PR DESCRIPTION
**Pull request by wltan**
- Elaborated on the `Model` component documentation. In particular, the `Transaction` class and its associated data fields.
- Added pitest to the DevOps guide
  - Fixed a minor typo in the DevOps guide
- Added an assertion and logging call

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -96,10 +96,37 @@ Below is the Sequence Diagram for interactions within the `Logic` component for
...
+The *data fields* `Title`, `Amount`, `Date`, and `Category` are encapsulations of an underlying Java data type.
+
+| Data Field  | Underlying Java Data Type   |
+| ----------- | -------------------------   |
+| `Title`     | `java.lang.String`          |
+| `Amount`    | `java.math.BigDecimal`      |
+| `Date`      | `java.time.LocalDate`       |
+| `Category`  | `java.lang.String`          |
+
+The underlying Java data types allow more operations to be done on `Transaction` objects, such as filtering transactions by date, or aggregating the amounts of expenses and incomes.
```
Comment (wltan) [inline docs/DeveloperGuide.md:118]: Unsure if the keywords e.g. "transaction", "date", "amount", "expense", "income" should reference the class names.

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java [hunk 2]
```diff
@@ -26,6 +26,7 @@ public Amount(String amount) {
         requireNonNull(amount);
         checkArgument(isValidAmount(amount), MESSAGE_CONSTRAINTS);
         value = new BigDecimal(amount.replaceFirst("^\\$", ""));
+        assert value.compareTo(BigDecimal.ZERO) >= 0; // amount should be non-negative
```
Comment (ianyong) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java:29]: Create an issue for this lest we forget to update the actual input validation rules?

**Review by ianyong — APPROVED** [review]
LGTM!

**Review by siddarth2824 — APPROVED** [review]
LGTM!

**Pull request merged**